dbms.py

```diff
@@ -109,3 +121,38 @@ def isBCNF(relation):
         return False
 
     return every(relation.fdSet.leftSideSet(), lambda lhs: relation.validKey(lhs) )
+
+def minimalCover(attributes, fdString):
+    fd_set = FunctionalDependencySet([ fd for fd in fdString.split(',')])
+    
+    #Split the dependencies
+    temp_fds = set((lhs,ele) for lhs, rhs in fd_set for ele in rhs)
+    fd_set.replaceItems(temp_fds) 
+    
+    #Reduce Right side
+    temp_fds = fd_set.duplicate()
+    for lhs, rhs in fd_set:
+        temp_fds.remove(lhs,rhs)
+        if set(rhs).issubset(temp_fds.closureSet(lhs)):
+            continue
+        else:
+            temp_fds.add(lhs,rhs)
+    #fd_set.replaceItems(temp_fds)
+    fd_set = temp_fds.duplicate()
+    
+    #Reduce Left side
+    temp_fds = fd_set.duplicate()
+    for lhs, rhs in fd_set:
+        if len(lhs) < 2: continue
+        temp_fds.remove(lhs,rhs)
+        temp_lhs = lhs
+        lhs_closure = fd_set.closureSet(lhs)
+        for ele in temp_lhs:
+            if lhs_closure == temp_fds.closureSet(ele):
+                lhs = e
```

Comment:
this should be `lhs = ele`, a little typo there

Reply:
yes.